src/models/models.jl

```diff
@@ -245,6 +245,12 @@ function infinite_terms(opsum::OpSum; kwargs...)
   # stores all terms starting on site `i`.
   opsum_cell_dict = groupreduce(minimum ∘ ITensors.sites, +, opsum)
   nsites = maximum(keys(opsum_cell_dict))
+  # check that we don't have terms we will ignore
+  dropped = filter(x -> x <= 0, keys(opsum_cell_dict))
```

Comment:
I don't remember all of the details of this code, is this checking if there is a term that has a site that is less than 1? Maybe that should just be an error, it seems weird to have terms that aren't valid that we just ignore.

Reply:
A few lines down you have `opsum_cell = [opsum_cell_dict[j] for j in 1:nsites]` where it is hardcoded to do sites 1 to nsites. Happy to switch this to an error and just enforce a sensible unit cell definition

Reply:
Yes an error would be better.